### DISCUSSION TITLE
Using Rainbowkit with React (non Nextjs) compatible authentication

### DISCUSSION THREAD

**Discussion by 0xAskar** [rainbow-me/rainbowkit#1426] [Q&A] [answered]
Are there any examples of using sign in with ethereum and non-nextAuth authentication for just react apps? For instance, I'm building a react browser extension that is not in NextJS, but wanted to still use rainbowkit for signing in and authentication. Would love to know if anyone has any ideas or examples for past usecases. Would assume someone has already done so.

**Comment by DanielSinclair** [comment, marked as answer]
@0xAskar We don't have a direct example available for this today, unfortunately. SIWE requires a backend server (for signature request generation and validation), so the Next.js frontend-backend coupling simplifies this implementation. However our Custom Authentication API is flexible enough to support a separate backend for these needs. Reference our [adapter docs](https://www.rainbowkit.com/docs/custom-authentication) for an example, and Spruce's example implementation of an Express backend [here](https://github.com/spruceid/siwe-quickstart/blob/main/02_backend/src/index.js).
